doc/customizing/client-side-config.md

```diff
@@ -174,3 +175,28 @@ Configuration of items to appear in the search bar
   ]
 }
 ```
+
+***
+
+### CatalogIndex
+
+If your TerriaMap has many (>50) dynamic groups (groups which need to be loaded - for example CKAN, WMS-group...) it may be worth generating a static catalog index JSON file. This file will contain ID, name and description fields of all catalog items, which can be used to search through the catalog very quickly without needing to load dynamic groups.
+
+The https://github.com/nextapps-de/flexsearch library is used to index and search the catalog index file.
+
+To generate the catalog index:
+
+- `npm run build-tools`
+- `node .\build\generateCatalogIndex.js config-url base-url` where
```

Comment:
It failed to run when using node 10.x (which is outdated now I guess) due to `TextEncoder` not being defined by `jsdom`. But would be good to leave a note to use at least 12.x.

Reply:
done!